packages/react/src/components/AssessShared/EnterEmailModal.tsx

```diff
@@ -0,0 +1,92 @@
+import { Box, Button, FormField, Input, Modal, SpaceBetween } from "@awsui/components-react";
+import { MemoryHistory } from "history";
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+interface Props {
+  visible: boolean;
+}
+
+// This function simply checks if the email is currently set
+// It is designed to be used when implementing the EnterEmailModal
+// However, it does not have to be used if the implementer
+// wants to use their own functionality
+// Use it to determine whether or not to display the modal
+export const isEmailSet = () => {
+  const email = window.electron.getState("email");
+  if (email === "" || email == null) {
+    return false;
+  }
+  return true;
+};
+
+export const EnterEmailModal: React.FC<Props> = React.memo(({ visible }) => {
+  const [isVisible, setVisible] = React.useState(visible);
+  const [emailValue, setEmailValue] = React.useState("");
+  const [emailError, setEmailError] = React.useState("");
+  const history = useHistory() as MemoryHistory;
+
+  // Used to check if the email string is valid, e.g. example@amazon.com
+  const validEmail = new RegExp(`^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$`);
```

Comment:
Does this regexp cover all cases of email addresses? Can we also check what are the best practices when validating email user input?

Reply:
Based on the official criteria mentioned here: https://medium.com/hackernoon/the-100-correct-way-to-validate-email-addresses-7c4818f24643

the regex matches all technically legal addresses except for dotless emails (discouraged by ICANN) and emails with whitespace. Best practice is to send a verification email to cover all cases, but I believe that may present an additional hurdle to the user. 

I went with this approach because it is simple and should allow 99% of emails. Developers, the primary users of our product, are likely to have typically formatted emails that will be allowed by this regex. 

I think our choices here are to 
a) Acknowledge that we will not accept a very small minority of emails, and maybe disclose that
b) Build additional functionality to send verification links to account for all cases

What do you think?